### PR TITLE
Added end to end test for mem ballast

### PR DESCRIPTION
**Pull request by owais**
This commit adds an end to end test to verify performance gain acheived
with memory ballast

It mainly checks the CPU usage and does not test against memory usage.
I'll be submitting another PR that will specifically test memory
behaviour and ensure the ballast does not actually consume memory.

`TestNoBackend10kSPS` test was very reliably consuming >20 CPU on my
machine so I increased the accepted values to 30 for this test.

### CODE REVIEW

File: testbed/tests/perf_test.go [hunk 1]
```diff
@@ -188,3 +158,81 @@ func Test1000SPSWithAttributes(t *testing.T) {
 		})
 	}
 }
+
+
+func Test1000SPSWithAttributes(t *testing.T) {
```
Comment (tigrannajaryan) [inline testbed/tests/perf_test.go]: This uses the same expectedMaxCPU as the one without ballast. However test results that you posted show that is actually uses less CPU. So, perhaps tighten a bit expectedMaxCPU in this test? I suggest you leave about 30% of slack in CPU (that was my rule of thumb when writing other perf tests).

File: testbed/tests/results/TESTRESULTS.md [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
+TestNoBackend10kSPS                     |PASS  |     10s|    49.9|    62.0|         28|         39|     98190|             0
+Test1000SPSWithAttributes/0*0bytes      |PASS  |     12s|    13.8|    23.0|         29|         40|     10000|         10000
+Test1000SPSWithAttributes/100*50bytes   |PASS  |     12s|    73.4|    82.0|         34|         48|     10000|         10000
+Test1000SPSWithAttributes/10*1000bytes  |PASS  |     12s|    78.2|    92.0|         32|         44|      9990|          9990
+Test1000SPSWithAttributes/20*5000bytes  |PASS  |     12s|   114.9|   119.9|         48|         67|      9970|          9970
+TestBallast1000SPSWithAttributes/0*0bytes|PASS  |     12s|    21.9|    25.3|         73|        114|     10000|         10000
+TestBallast1000SPSWithAttributes/100*50bytes|PASS  |     12s|    60.2|    68.3|        498|        847|     10000|         10000
+TestBallast1000SPSWithAttributes/10*1000bytes|PASS  |     12s|    41.0|    45.3|        354|        576|      9990|          9990
+TestBallast1000SPSWithAttributes/20*5000bytes|PASS  |     12s|    64.4|    74.0|        780|       1079|      9990|          9990
```
Comment (tigrannajaryan) [inline testbed/tests/results/TESTRESULTS.md]: This is a nice CPU improvement.

File: testbed/tests/perf_test.go (hunk 1 above)
Comment (owais) [reply testbed/tests/perf_test.go]: I think I did reduce the numbers before creating the PR. Probably I missed something in the first push. In any case, updated the numbers and reduced them a further.

**Comment by tigrannajaryan** [comment]
@owais  Build status is not reported from Travis. You may want to restart.

**Comment by owais** [comment]
Looks like a bug. The build has passed: https://travis-ci.org/open-telemetry/opentelemetry-service/builds/552958842?utm_source=github_status&utm_medium=notification

I don't have the option to restart builds for this repo and it's PRs.

**Comment by tigrannajaryan** [comment]
It will restart automatically after you merged the other PR and rebase this one.

**Comment by tigrannajaryan** [comment]
Failed because of https://github.com/open-telemetry/opentelemetry-service/issues/43

**Comment by owais** [comment]
Yea, I've been seeing quite a few random failures. Will rebase after we merge #85

File: testbed/tests/results/TESTRESULTS.md [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
+----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:
+TestIdleMode                            |PASS  |     10s|     0.3|     0.7|         24|         32|         0|             0
+Test10kSPS                              |PASS  |     17s|    73.3|    78.3|         34|         42|    148760|        148760
+TestNoBackend10kSPS                     |PASS  |     10s|    79.5|    81.7|         29|         40|     98770|             0
+Test1000SPSWithAttributes/0*0bytes      |PASS  |     12s|    25.8|    26.7|         30|         41|     10000|         10000
+Test1000SPSWithAttributes/100*50bytes   |PASS  |     12s|    87.1|    93.7|         34|         47|      9990|          9990
+Test1000SPSWithAttributes/10*1000bytes  |PASS  |     12s|    67.8|    74.3|         32|         44|     10000|         10000
+Test1000SPSWithAttributes/20*5000bytes  |PASS  |     12s|    96.7|   102.0|         46|         71|     10000|         10000
+TestBallast1000SPSWithAttributes/0*0bytes|FAIL  |      4s|    25.3|    25.3|         41|         82|      2830|          2790
```
Comment (tigrannajaryan) [inline testbed/tests/results/TESTRESULTS.md]: Was this supposed to fail?

Comment (owais) [reply testbed/tests/results/TESTRESULTS.md]: No. Let's merge #85 first and then I'll rebase this which'll fix it.

**Comment by tigrannajaryan** [comment]
@owais did you make changes or just rebased? If you need to rebase it is better to use "Update branch" button in Github. It reruns CI, but does not invalidate reviews (unlike rebasing and pushing the branch manually, which requires re-approval). So you don't need to ask for reapproval in that case.

**Comment by owais** [comment]
@tigrannajaryan I rebased. Update branch features ends up creating a merge commit as it merges master into the branch. Is that acceptable?

File: testbed/tests/results/TESTRESULTS.md (hunk 3 above)
Comment (owais) [reply testbed/tests/results/TESTRESULTS.md]: Fixed.

**Comment by tigrannajaryan** [comment]
>Update branch features ends up creating a merge commit as it merges master into the branch. Is that acceptable?

@owais I don't think this is true. If I remember correctly when you merge the PR, the commit is squashed and rebased, no merge commit is created.

**Comment by pjanotti** [comment]
CI hit https://github.com/open-telemetry/opentelemetry-service/issues/59

**Comment by owais** [comment]
> @owais I don't think this is true. If I remember correctly when you merge the PR, the commit is squashed and rebased, no merge commit is created.

I see. I meant it creates a  merge commit in the branch when updating from master but that wouldn't matter if the repo has only "squash and merge" available. 

Update again. Thanks.

**Pull request merged**